### ISSUE TITLE
[P2] 被剔除但已執行的 forced target 被當作 witness，誤報污染並中止 deviation 評估

### ISSUE THREAD

**Issue by HHim8826** [HHim8826/Mortal#20]
審查範圍：`0833528` 的錯配 logits 過濾，在 `61f07a8ab0145128f309d5ef4902e9e23316a3ab` 確認。

修正 #12 時，程式會把「已觸發，但實際 argmax 不符合 metadata」的 target 從 `targets` 刪掉。後面的統計卻把所有不在 `targets` 的牌譜視為「完全沒有 fork 的 witness」。這些被剔除的牌譜可能早已強制改過動作，其差異不能算成其他遊戲造成的污染。

位置：
- [fired mismatch 從 targets 刪除](https://github.com/HHim8826/Mortal/blob/61f07a8ab0145128f309d5ef4902e9e23316a3ab/mortal/deviation_cost.py#L405-L413)
- [不在 targets 就計入 witness](https://github.com/HHim8826/Mortal/blob/61f07a8ab0145128f309d5ef4902e9e23316a3ab/mortal/deviation_cost.py#L426-L437)
- [污染率超標即中止](https://github.com/HHim8826/Mortal/blob/61f07a8ab0145128f309d5ef4902e9e23316a3ab/mortal/deviation_cost.py#L474-L480)

### 重現條件

例如原本 pass=45 的決策，錯配 metadata 指向 argmax=41；以新的 `--force argmax` 對照模式強制動作 41：

1. Forcer 已執行 45 → 41，兩份牌譜因此不同。
2. argmax mismatch 檢查正確識別錯配，將該 target 剔除。
3. 統計階段反而把這份受干預的牌譜算為 changed witness。

離線重現直接執行此 commit 的過濾、witness 分類及中止判斷，輸入一份真正未干預且完全相同的牌譜，以及上述一份已干預後剔除的牌譜：

```text
真正未干預：1；其中有差異：0
程式計算 witness：2；changed：1
結果：50% > 預設 10%，觸發 SystemExit
```

錯誤訊息會要求關閉 AMP 或減少 arena 遊戲數，但這個例子的差異由指定干預直接造成，和跨遊戲浮點污染無關。

### 建議

把「沒有干預」、「已干預但測量無效」、「有效 target」分開。被剔除且已 fired 的項目應保留 rejected 狀態，排除於 witness 分子與分母；不要以是否仍留在 targets 判斷有無干預。

本項是新增過濾流程的分類錯誤，並非再次回報 #12 的 metadata 對齊根因。驗證使用實際控制流程及合成 log pair，未聲稱執行完整 GPU arena。

**Comment by HHim8826** [comment]
Fixed in de9c8ee. The set of hanchans that were armed is kept, and one that is armed and later dropped is counted as rejected rather than as a witness — it is excluded from both the numerator and the denominator, and reported separately.

Your reading also corrects the published result, in the direction that makes the method look better than I reported it. The last run said '1,001 identical / 2 changed, 0.20%'. The design has exactly 1,000 witnesses (10,000 hanchans, one in ten), so the denominator of 1,003 was the three dropped targets. The two that changed must be the two that 'never came round' — those diverged before their own decision, which is *why* they were dropped, so their logs necessarily differ — and there were only two changed in total. So among hanchans nothing was done to: **0 of 1,000 changed**. Cross-game perturbation is real but its only evidence is 2 of 9,000 forked games losing their target, which was already reported separately. The write-up has been corrected.
